content/md/en/docs/reference/frame-macros.md

```diff
@@ -1,118 +1,206 @@
+To learn more about the FRAME macros used in the Substrate runtime, you can install and use [cargo-expand](https://docs.rs/crate/cargo-expand).
+After you install cargo-expand, you can use the `cargo expand` command to display the results of the code contained in the macros the runtime uses.
 
-**Macros in the FRAME support library**
+## FRAME support and system macros
 
-- in `frame_support`:
-  - [function-like](https://paritytech.github.io/substrate/master/frame_support/index.html#macros) macros
-  - [derive](https://paritytech.github.io/substrate/master/frame_support/index.html#derives) macros
-  - [attribute](https://paritytech.github.io/substrate/master/frame_support/index.html#attributes) macros
+Substrate primitives and FRAME both rely on a collection of different macros.
+This section provides an overview of the macros provided in the FRAME support and FRAME system libraries.
+In most cases, these macros provide the framework that other pallets depend on and you should be familiar with how and where they are used in the runtime logic.
+After the overview, this section describes the specific macros that you are most likely to use as a runtime developer. 
 
-**Macros in the FRAME system library**
+### FRAME support macros
 
-- in `sp_core`:
-  - [function-like macros](https://paritytech.github.io/substrate/master/sp_core/index.html#macros) macros
-  - [derive `RuntimeDebug`](https://paritytech.github.io/substrate/master/sp_core/index.html#derives) macro
-- in `sp_runtime`:
-  - [function-like](https://paritytech.github.io/substrate/master/sp_runtime/index.html#macros) macros
-  - [derive](https://paritytech.github.io/substrate/master/sp_runtime/index.html#derives) macros
-- in `serde`: [function-like](https://paritytech.github.io/substrate/master/sp_runtime/index.html#derives) macros
-- in `sp_api`: [function-like](https://paritytech.github.io/substrate/master/sp_api/index.html#macros) macros
-- in `sp_std`: [function-like](https://paritytech.github.io/substrate/master/sp_std/index.html#macros) macros
-- in `sp_version`: [function-like](https://paritytech.github.io/substrate/master/sp_version/index.html#macros) macros
+The `frame_support` crate provides many of the most important declarative, derive, attribute-like, and function-like macros used in the runtime.
+A few of the important macros that you should be familiar with from the `frame_support` crate include the following:
 
-See the `dependencies` in the `Cargo.toml` file of the node template runtime to see where these macros are used.
+- `construct_runtime`	used to construct runtime from the list of pallets you have implemented.
+- `decl_error` used to declare an error type for a pallet in the runtime.
+- `decl_event`	used to implement event for a pallet in the runtime.
+- `decl_module`	used to declare a Module data structure and a Call enumeration for a dispatch.
+- `decl_storage` used to declare strongly-typed wrappers around codec-compatible types in storage.
```

Comment:
we should remove all references to these `decl_*` macros

Reply:
that one is okay :)

The FRAME v1 macros are all declarative, whereas the frame v2 are all attribute macros.

Beyond that, just the ones that use `decl_xx` instead of `#[pallet::xx]`